packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -206,23 +206,39 @@ export class OdspDocumentService implements IDocumentService {
                 ? Promise.resolve(null)
                 : this.getWebsocketToken!(options);
             const joinSessionPromise = this.joinSession(requestWebsocketTokenFromJoinSession).catch((e) => {
-                let code: string | undefined;
+                const stack: any = [];
                 try {
-                    code = e?.response ? JSON.parse(e?.response)?.error?.code : undefined;
+                    if(e !== undefined) {
```

Comment:
This logic would need to be used in other places, so we are better of refactor it into stand-alone function, or put this logic directly into createOdspNetworkError to put codes into a property on error object.
Later has some benefits as we may chose to log all or one of them (the most inner) code into telemetry as a separate field, in addition to already logging full response, as it allows much easier search / grouping and is not susceptible to problems described in https://github.com/microsoft/FluidFramework/issues/6139.

Reply:
Super excited to get clarity on ODSP error response format! @chensixx I would like to sync up with you about what Vlad's saying here.